src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,153 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final Object[] EMPTY_ELEMENTDATA = {};
+    private Object[] elementData;
+    private int size;
+
+    public ArrayList(int initialCapacity) {
+        if (initialCapacity > 0) {
+            this.elementData = new Object[initialCapacity];
+        } else if (initialCapacity == 0) {
+            this.elementData = EMPTY_ELEMENTDATA;
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("Illegal Capacity: "
+                    + initialCapacity);
+        }
+    }
+
+    public ArrayList() {
+        this.elementData = EMPTY_ELEMENTDATA;
+    }
+
     @Override
     public void add(T value) {
-
+        add(value, elementData, size);
     }
 
     @Override
     public void add(T value, int index) {
+        rangeCheckForAdd(index);
+        final int s = size;
+        Object[] elementDataCopy = this.elementData;
```

Comment:
delete **s** and **elementDataCopy** 
![image](https://user-images.githubusercontent.com/20326858/225312354-d88d094f-c247-4042-b6db-e28a9e9575b7.png)